docs/events.md

```diff
@@ -0,0 +1,696 @@
+      * Auto-delete: `false`
+      * vhost:
+    * Routing key: device.unregistered
+
+</details>
+
+### **device.schema.updated** <a name="device-schema-updated"></a>
+
+Event that represents a thing's schema was updated.
+
+<details>
+  <summary>Payload</summary>
+
+  JSON in the following format:
+
+  * `id` **String** thing's ID
+  * `error` **String** a string with detailed error message
+
+  Success example:
+
+  ```json
+  {
+    "id": "fbe64efa6c7f717e",
+    "error": null
+  }
+  ```
+
+  Error example:
+
+  ```json
+  {
+    "id": "3aa21010cda96fe9",
+    "error": "- \"valueType\" must be larger than or equal to 1\n- \"name\" must be a string\n- \"value\" does not contain 1 required value(s)"
+  }
+  ```
+</details>
+
+<details>
+  <summary>Binding</summary>
+
+  * AMQP:
+    * Exchange:
+      * Type: direct
+      * Name: device
+      * Durable: `true`
+      * Auto-delete: `false`
+      * vhost:
+    * Routing key: device.schema.updated
+
+</details>
+
+### **data.published** <a name="data-published"></a>
+
+Event that represents a data published from a thing's sensor.
+
+<details>
+  <summary>Payload</summary>
+
+  JSON in the following format:
+
+  * `id` **String** thing's ID
+  * `data` **Array** data items to be published, each one formed by:
+    * `sensorId` **Number** sensor ID
+    * `value` **Number|Boolean|String** sensor value
+
+  Example:
+
+  ```json
+  {
+    "id": "fbe64efa6c7f717e",
+    "data": [
+      {
+        "sensorId": 1,
+        "value": false
+      },
+      {
+        "sensorId": 2,
+        "value": 1000
+      }
+    ]
+  }
+  ```
+</details>
+
+<details>
+  <summary>Binding</summary>
+
+  * AMQP:
+    * Exchange:
+      * Type: fanout
+      * Name: data.published
+      * Durable: `true`
+      * Auto-delete: `false`
+      * vhost:
+
+</details>
+
+### **device.[id].data.request** <a name="device-<id>-data-request"></a>
+
+Event-command to request a specific thing's sensor data after validated.
+
+<details>
+  <summary>Headers</summary>
+
+  * `token` **String** user's token
+
+</details>
+
+<details>
+  <summary>Payload</summary>
+
+  JSON in the following format:
+
+  * `id` **String** thing's ID
+  * `data` **Array** data items to be published, each one formed by:
+    * `sensorId` **Number** sensor ID
+    * `value` **Number|Boolean|String** sensor value
+
+  Example:
+
+  ```json
+  {
+    "id": "fbe64efa6c7f717e",
+    "data": [
+      {
+        "sensorId": 1,
+        "value": false
+      },
+      {
+        "sensorId": 2,
+        "value": 1000
+      }
+    ]
+  }
+  ```
+</details>
+
+<details>
+  <summary>Binding</summary>
+
+  * AMQP:
+    * Exchange:
+      * Type: direct
+      * Name: device
+      * Durable: `true`
+      * Auto-delete: `false`
+      * vhost:
+    * Routing Key: `device.<id>.data.request`
+
+</details>
+
+### **device.[id].data.update** <a name="device-<id>-data-update"></a>
+
+Event-command to update a specific thing's sensor data after validated.
```

Comment:
Could you be a little more specific on what is `validated`? It might get confusing for newcomers :stuck_out_tongue:

Reply:
Done.